src/Error/ExceptionRenderer.php

```diff
@@ -119,8 +97,14 @@ protected function _getErrorData()
                 'class' => get_class($viewVars['error']),
                 'code' => $viewVars['error']->getCode(),
                 'message' => $viewVars['error']->getMessage(),
-                'trace' => preg_split('@\n@', $viewVars['error']->getTraceAsString()),
             ];
+
+            if (!isset($data['trace'])) {
```

Comment:
Since 3.1's exception renderer already sets [trace](https://github.com/cakephp/cakephp/blob/master/src/Error/ExceptionRenderer.php#L169) it's present in `$data['trace']` unless you have a custom method for an exception. So I set it under `$data` itself to maintain consistency.